File: tarea?/ejercicio2.py
import time
from pathlib import Path

filename = Path("/home/arelyxl/Downloads/VsCode/Operating_system/Shell/tarea?/hw.txt").resolve()  # Usamos .resolve() para obtener la ruta absoluta

print(f"Esperando a que se cree el archivo {filename}...")

while not filename.is_file():  
    print("Archivo no encontrado. Reintentando en 1 segundo...")
    time.sleep(1)

print(f"{filename.name} encontrado. Continuando con el procesamiento.")

with open(filename, mode="r") as file:
    print("Contenido del archivo:")
    print(file.read())

File: tarea?/ejercicio2.cpp
#include <iostream>
#include <fstream>
#include <thread>
#include <chrono>
#include <filesystem>

using namespace std;

bool existFile(string name);

int main(){
    do{
        cout << "No se creo el archivo hw.txt" << endl;
        this_thread::sleep_for(chrono::seconds(1));
        cout << "Buscando en: " << std::filesystem::current_path() << endl;
    }while(! existFile("/home/arelyxl/Downloads/VsCode/Operating_system/Shell/tarea?/hw.txt"));
    cout << "El archivo hw.txt fue creado" << endl;
}

bool existFile(string name){
    ifstream file(name);
    return file.good();
}